src/lib/operators/projection.cpp

```diff
@@ -272,10 +272,12 @@ std::shared_ptr<const Table> Projection::_on_execute() {
       std::vector<SortColumnDefinition> transformed;
       transformed.reserve(sorted_by.size());
       for (const auto& [column_id, mode] : sorted_by) {
-        if (!input_column_to_output_column.count(column_id)) {
+        const auto iter = std::find_if(output_column_to_input_column.begin(), output_column_to_input_column.end(),
+                                       [column_id = column_id](const auto entry) { return column_id == entry.second; });
+        if (iter == output_column_to_input_column.end()) {
           continue;  // column is not present in output expression list
         }
-        const auto projected_column_id = input_column_to_output_column[column_id];
+        const auto projected_column_id = iter->first;
```

Comment:
Where is this tested?

Reply:
Tested with extended test in the projection tests.